### DISCUSSION TITLE
How to setup cache storage to change in runtime ?

### DISCUSSION THREAD

**Discussion by geminigeek** [nitrojs/nitro#2944] [Q&A] [answered]
hi,

this is regarding this section https://nitro.build/guide/cache#caching-route-rules , i am using nuxt 3.
how can i put the storage config in runtime key to dynamically change redis server this `url: "redis://localhost:6379"`, in run time ? 

example from the link

```
export default defineNuxtConfig({
  nitro: {
    storage: {
      redis: {
        driver: "redis",
        url: "redis://localhost:6379",
      },
    },
  },
  routeRules: {
    "/blog/**": { cache: { maxAge: 60 * 60, base: "redis" } },
  },
});
```

**Comment by cjpearson** [comment, marked as answer]
Instead of defining the storage in the nuxt.config, you can create a nitro plugin which mounts a storage based on some runtime configuration.

https://github.com/nitrojs/nitro/issues/1161#issuecomment-1511444675

Something like this:

```ts
import redis from "unstorage/drivers/redis";

export default defineNitroPlugin(() => {
  const storage = useStorage();
  const config = useRuntimeConfig();
  storage.mount(
    "/redis",
    redis({
      url: config.redisUrl
    })
  );
});
```

```ts
export default defineNuxtConfig({
  runtimeConfig: {
    redisUrl: '' // set in .env with NUXT_REDIS_URL
  },
  routeRules: {
    "/blog/**": { cache: { maxAge: 60 * 60, base: "redis" } },
  },
});
```

**Reply by geminigeek** [reply]
thanks, i implemented like this only
